src/components/MarketsTable.tsx

```diff
@@ -0,0 +1,149 @@
+import { Button, Flex, Stack, Tag, useMediaQuery } from '@chakra-ui/react'
+import { bnOrZero } from '@shapeshiftoss/chain-adapters'
+import type { Asset } from '@shapeshiftoss/types'
+import { truncate } from 'lodash'
+import { memo, useCallback, useMemo } from 'react'
+import { RiArrowRightDownFill, RiArrowRightUpFill } from 'react-icons/ri'
+import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router'
+import type { Column, Row } from 'react-table'
+import { Amount } from 'components/Amount/Amount'
+import { Display } from 'components/Display'
+import { ReactTableNoPager } from 'components/ReactTable/ReactTableNoPager'
+import { AssetCell } from 'components/StakingVaults/Cells'
+import { Text } from 'components/Text'
+import { useInfiniteScroll } from 'hooks/useInfiniteScroll/useInfiniteScroll'
+import { SparkLine } from 'pages/Buy/components/Sparkline'
+import { selectMarketDataUserCurrency } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+import { breakpoints } from 'theme/theme'
+
+const arrowUp = <RiArrowRightUpFill />
+const arrowDown = <RiArrowRightDownFill />
+
+const paddingX = { base: 4, md: 0 }
+
+type RowProps = Row<Asset>
+
+type MarketsTableProps = {
+  rows: Asset[]
+  onRowClick: (arg: Row<Asset>) => void
+}
+
+export const MarketsTable: React.FC<MarketsTableProps> = memo(({ rows, onRowClick }) => {
+  const translate = useTranslate()
+  const history = useHistory()
+  const [isLargerThanMd] = useMediaQuery(`(min-width: ${breakpoints['md']})`, { ssr: false })
+  const marketPrices = useAppSelector(selectMarketDataUserCurrency)
```

Comment:
```suggestion
  const marketDataById = useAppSelector(selectMarketDataUserCurrency)
```

Reply:
Also @reallybeard if you wish to, cells are components, so we can `selectMarketDataById` on them instead of selecting all above which will be less re-rendery